proposals/nnnn-Use-Taskmaster-To-Handle-Queuing-Operations-In-Managers.md

```diff
@@ -0,0 +1,80 @@
+
+[`TaskMaster`](https://github.com/livio/Taskmaster) is queue oriented thread management solution developed by the SDL project maintainer to solve aforementioned issue. It is designed in a way that allows each manager to have its own queue of operations but still run all queues on a fixed number of threads. This gets rid of the thread-per-queue restriction that the `ThreadPoolExecutor` solution suffers. SDL iOS already uses `NSOperationQueue` to manage queuing operations in the managers which is a similar solution but it handles threads at the OS level.
+
+The proposal suggests adding an API called `getTaskmaster()` to the `Isdl` interface which allows all managers to access the same `Taskmaster` instance:
+
+```
+public interface ISdl {
+    ...
+
+    Taskmaster getTaskmaster();
+}
+```    
+
+Each manager that uses queues can create its own queue using the `Taskmaster` instance and add operations to the queue to be executed it order:
+
+```
+class ChoiceSetManager {
+    Queue operationQueue;
+
+    ChoiceSetManager (){
+        operationQueue = internalInterface.getTaskmaster().createQueue("operationQueue", 1, false);
+        ...
+    }
+
+    void preloadChoices(@NonNull List<ChoiceCell> choices, @Nullable final CompletionListener listener) {
+        ...
+        PreloadChoicesOperation preloadChoicesOperation = new PreloadChoicesOperation("preloadChoicesOperation");
+        operationQueue.add(preloadChoicesOperation, false);
+        ...
+    }
+}
+```
+
+Each operation can extend `Task` class from [`TaskMaster`](https://github.com/livio/Taskmaster) and implement the operation logic in the `onExecute()` callback. [`TaskMaster`](https://github.com/livio/Taskmaster) will take care of running that logic when it finishes running all previous operations in the queue:
+
+```
+class PreloadChoicesOperation extends Task {
+    @Override
+    public void onExecute() {
+        // Do work here
+    }
+}
+```
+
+## Potential downsides
+
+This proposal will add a dependency to the SDL Java Suite library. However, because the dependency is created and managed by the SDL Project maintainer, any updates or compatibility issues should be made as needed.
```

Comment:
```suggestion
This proposal will add a dependency to the SDL Java Suite library. However, because the dependency is created and managed by the SDLC Project Maintainer, any updates or compatibility issues should be made as needed.
```